extensions/cache/runtime/src/main/java/io/quarkus/cache/runtime/caffeine/CaffeineCacheBuildRecorder.java

```diff
@@ -19,14 +21,15 @@ public void buildCaches(BeanContainer beanContainer, Set<CaffeineCacheInfo> cach
         // The number of caches is known at build time so we can use fixed initialCapacity and loadFactor for the caches map.
         Map<String, CaffeineCache> caches = new HashMap<>(cacheInfos.size() + 1, 1.0F);
 
+        ManagedExecutor managedExecutor = Arc.container().instance(ManagedExecutor.class).get();
```

Comment:
Can't you use the `BeanContainer` since you pass it?

Reply:
Good catch. I tried a few build time solutions based on `BeanContainer` that didn't work, but I forgot to try it from the recorder.